src/service/custom-element-registry.js

```diff
@@ -142,6 +143,18 @@ export function registerElement(win, name, implementationClass) {
   win['customElements'].define(name, klass);
 }
 
+/**
+ * Registers a new Bento-based custom element with its implementation class.
+ * @param {!Window} win The window in which to register the elements.
+ * @param {string} name Name of the custom element
+ * @param {typeof ../base-element.BaseElement} implementationClass
+ * @throws {UserError} if the required Bento experiment is not enabled
+ */
+export function registerBentoElement(win, name, implementationClass) {
```

Comment:
This causes the error to be thrown at registration rather than during layout. Is this problematic?

Reply:
Yes, runtime cannot handle that, and will make all elements pending.

Reply:
Seems like `PreactBaseElement.isLayoutSupported` could maybe do this

Reply:
`isLayoutSupported` is called per element, and a thrown error there will turn off the element in Resources. But the other elements will be unaffected.